### DISCUSSION TITLE
Trying to build an ANGLE formula

### DISCUSSION THREAD

**Discussion by computermouth** [Homebrew/discussions#4652] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

Just some questions really. I'm trying to write a formula for [ANGLE](https://github.com/google/angle/tree/main), it's a library which can translate various graphics APIs to a more native-friendly API (I've written a GLES3 game, and would like to build for mac).

Disclaimer: I'm not normally a mac user, I've used homebrew a handful of times, but I've packaged software for several Linux distros before.

Getting to the questions:
1. Is there a way to set build dependencies? All the build tools seem to be invoked as `system "make install"`, but how do I know what's on the build system?
2. I imagine using `git` in the build process is unlikely to pass a vetting, but ANGLE requires Google's [depot_tools](https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up) (shell, batch, and python scripts) for it's builds. But these seem like pretty weird and specific tools that wouldn't warrant an additional package. And even if it did, again, question 1, how do I set build dependencies?
3. Assuming there is no build dependencies separation from normal dependencies, what would be the suggested path? Should I indeed just make a depot_tools package, and I guess just only add the python scripts which I know are used in the build?

**Comment by SMillerDev** [comment, marked as answer]
> Is there a way to set build dependencies? All the build tools seem to be invoked as system "make install", but how do I know what's on the build system?

`depends_on "stuff" => :build`

> I imagine using git in the build process is unlikely to pass a vetting, but ANGLE requires Google's depot_tools (shell, batch, and python scripts) for it's builds. But these seem like pretty weird and specific tools that wouldn't warrant an additional package. And even if it did, again, question 1, how do I set build dependencies?

Unless it's a very popular tool, I don't think we'd want that in Homebrew core indeed.

**Reply by alebcay** [reply]
Yes, `depot_tools` is maintained on a rolling basis and has no tags (see https://chromium.googlesource.com/chromium/tools/depot_tools.git). It would not be a good fit to include in homebrew-core.

**Comment by xlla** [comment]
I'v found this formula for [google ANGLE](https://github.com/knazarov/homebrew-qemu-virgl/releases/tag/libangle-20211212.1)

the depot_tools is crazy and it pull 14G+ code from net,  then run to the error and said it need 10.15+ SDK (I am on 10.13 )
Most of pulled is android develop ndk and sdk, it seems it want to build for android and extract some libs.

hope this can help you.
